app/src/androidMain/kotlin/de/westnordost/streetcomplete/ui/common/image_select/ImageList.kt

```diff
@@ -0,0 +1,131 @@
+package de.westnordost.streetcomplete.ui.common.image_select
+
+import android.content.res.Configuration
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import de.westnordost.streetcomplete.quests.bicycle_repair_station.BicycleRepairStationService
+import de.westnordost.streetcomplete.quests.bicycle_repair_station.asItem
+import kotlin.math.max
+
+@Composable
+fun <T> ImageList(
+    imageItems: List<ImageListItem<T>>,
+    onClick: (index: Int, item: ImageListItem<T>) -> Unit,
+    modifier: Modifier = Modifier,
+    itemsPerRow: Int = 4,
+    itemRole: Role = Role.Checkbox,
+    itemContent: @Composable (item: ImageListItem<T>, index: Int, onClick: () -> Unit, role: Role) -> Unit
```

Comment:
Regarding this interface:

What is `onClick` and `index` for? Is the composable just expected to pass it through? In that case, should just wrap it in a `Box` in line 31 and register the `Modifier.clickable {}` there. So, it could look like this:
`itemContent: @Composable (item: DisplayItem<T>, selected: Boolean) -> Unit`

Reply:
I'd say, arguably, this logic doesn't belong there, it belongs outside of the `ImageList`.

I.e.

```kotlin

ImageList(
  items = items,
  onSelected = { newItems ->
    if (plastic in newItems) showADialog(...)
    else items = newItems
  }  
)
```

Then again... I mean, this is how it usually should be done. But in this case, maybe that wouldn't work, as we actually want to respond to the item just having been clicked, not respond to the state... hm. Well, I guess the ramp part can be *very well* done with the approach I mentioned. But the recycling thing is more difficult.

Reply:
Only thing I can think of is pass a secondary argument of clickeditem that most implementations of onSelected ignore?